app/models/survey-model.js

```diff
@@ -18,14 +18,74 @@ if ( process.env.NODE_ENV === 'test' ) {
     client.select( 15 );
 }
 
+/**
+ * @typedef {import('./account-model').AccountObj} AccountObj
+ */
+
+/**
+ * @typedef {import('./account-model').EnketoRecord} EnketoRecord
+ */
+
+/**
+ * @typedef {typeof import('libxmljs').Document} XMLJSDocument
+ */
+
+/**
+ * @typedef {(doc: XMLJSDocument) => XMLJSDocument} EnketoTransformerPreprocess
+ */
+
+/**
+ * @typedef SurveyCredentials
+ * @property { string } user
+ * @property { string } pass
+ * @property { string } bearer
+ */
+
+/**
+ * @typedef SurveyExternalData
+ * @property { string } id
+ * @property { string } src
+ * @property { string | Document } xml
+ */
+/**
+ * @typedef SurveyInfo
+ * @property { string } downloadUrl
+ * @property { string } manifestUrl
+ */
+
 /**
  * @typedef SurveyObject
  * @property { string } openRosaServer
  * @property { string } openRosaId
  * @property { string } enketoId
  * @property { string } theme
- * @property { object } info
- * @property { string } info.hash
+ * @property { SurveyInfo } [info]
+ * @property { AccountObj } [account]
+ * @property { boolean | 'true' | 'false' } [active]
+ * @property { string } [cookie]
+ * @property { SurveyCredentials } [credentials]
+ * @property { string } [customParam]
+ * @property { SurveyExternalData[] } [externalData]
+ * @property { string } [form]
+ * @property { string } [formHash]
+ * @property { EnketoRecord } [instance]
+ * @property { Array<string | object> } [instanceAttachments]
+ * @property { string } [instanceId]
+ * @property { EnketoRecord } [lastSavedRecord]
+ * @property { Record<string, unknown> } [languageMap]
+ * @property { Record<string, unknown> } [manifest]
+ * @property { string } [model]
+ * @property { EnketoTransformerPreprocess } [preprocess]
+ * @property { string } [returnUrl]
+ * @property { string } [xslHash]
+ * @description
+ *   `SureveyObject` is Enketo's internal representation of an XForm, with some
+ *   additional properties representing resolved/deserialized external data.
+ *   This type definition captures the current state of "what is"—i.e. the full
+ *   known set of properties which may be added to a `SureveyObject` through
+ *   several data flow paths throught enketo-express. Some related resources,
+ *   notably those describing instances, are only populated in paths specific
+ *   to the interaction between a `SurveyObject` and those resources.
```

Comment:
1. While this is, I believe, descriptive of the various states a `Survey` might be in at various points in the data flow, it’s obviously not what one would consider a survey in the sense it might be understood as a Form representation. Late in the process of this PR, @lognaturel discussed strategies for gradually decoupling things that are attached to data types this way. The first such decoupling is removal of an `htmlView` property, which you’ll see further down in the PR.